ydb/core/wrappers/fs_storage.cpp

```diff
@@ -0,0 +1,518 @@
+        typename TEvResponse::TAwsResult awsResult;
+        Aws::Utils::Outcome<typename TEvResponse::TAwsResult, Aws::S3::S3Error> outcome(std::move(awsResult));
+
+        std::unique_ptr<TEvResponse> response;
+        if constexpr (HasKeyConstructor<TEvResponse>()) {
+            Y_ENSURE(key, "Key is required for this response type");
+            response = std::make_unique<TEvResponse>(*key, std::move(outcome));
+        } else {
+            response = std::make_unique<TEvResponse>(std::move(outcome));
+        }
+        this->Send(sender, response.release());
+    }
+
+    template<typename TEvResponse>
+    void ReplyError(const NActors::TActorId& sender, const std::optional<TString>& key, const TString& errorMessage) {
+        Aws::Client::AWSError<Aws::S3::S3Errors> awsError(
+            Aws::S3::S3Errors::INTERNAL_FAILURE,
+            "FsStorageError",
+            errorMessage,
+            false // not retryable for FS errors
+        );
+        Aws::S3::S3Error error(std::move(awsError));
+        Aws::Utils::Outcome<typename TEvResponse::TAwsResult, Aws::S3::S3Error> outcome(std::move(error));
+
+        std::unique_ptr<TEvResponse> response;
+        if constexpr (HasKeyConstructor<TEvResponse>()) {
+            Y_ENSURE(key, "Key is required for this response type");
+            response = std::make_unique<TEvResponse>(*key, std::move(outcome));
+        } else {
+            response = std::make_unique<TEvResponse>(std::move(outcome));
+        }
+        this->Send(sender, response.release());
+    }
+
+public:
+    TFsOperationActor(const TString& basePath)
+        : BasePath(basePath)
+    {
+    }
+
+    ~TFsOperationActor() {
+        CleanupActiveSessions();
+    }
+
+    void Bootstrap() {
+        FS_LOG_T("TFsOperationActor Bootstrap called");
+        Become(&TThis::StateWork);
+    }
+
+    void PassAway() override {
+        CleanupActiveSessions();
+        NActors::TActorBootstrapped<TFsOperationActor>::PassAway();
+    }
+
+private:
+    void CleanupActiveSessions() {
+        FS_LOG_T("TFsOperationActor: cleaning up"
+            << ": active MPU sessions# " << ActiveUploads.size());
+        for (auto& [uploadId, session] : ActiveUploads) {
+            try {
+                const TString filePath = session->Key;
+                session->File.Close();
+                NFs::Remove(filePath);
+
+                FS_LOG_T("TFsOperationActor: closed and deleted incomplete file"
+                    << ": uploadId# " << uploadId
+                    << ", file# " << filePath);
+            } catch (const std::exception& ex) {
+                FS_LOG_W("Failed to cleanup MPU session"
+                    << ": uploadId# " << uploadId
+                    << ", error# " << ex.what());
+            }
+        }
+        ActiveUploads.clear();
+    }
+
+public:
+
+    STATEFN(StateWork) {
+        FS_LOG_D("TFsOperationActor StateWork received event type"
```

Comment:
Имхо, это `FS_LOG_T`, т.к. каждый обработчик отдельно логируется с `FS_LOG_D`.
```suggestion
        FS_LOG_T("TFsOperationActor StateWork received event type"
```